app/src/main/java/com/example/encoratask/ItemDetailActivity.kt

```diff
@@ -16,34 +17,24 @@ import android.view.MenuItem
 class ItemDetailActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
+
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_item_detail)
         setSupportActionBar(findViewById(R.id.detail_toolbar))
 
-        findViewById<FloatingActionButton>(R.id.fab).setOnClickListener { view ->
-            Snackbar.make(view, "Replace with your own detail action", Snackbar.LENGTH_LONG)
-                    .setAction("Action", null).show()
-        }
+        // -- Get intent
+        val character = intent.getSerializableExtra(ItemDetailFragment.ARG_ITEM) as Character
 
-        // Show the Up button in the action bar.
-        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        // -- Init views
+        this.initViews()
 
-        // savedInstanceState is non-null when there is fragment state
-        // saved from previous configurations of this activity
-        // (e.g. when rotating the screen from portrait to landscape).
-        // In this case, the fragment will automatically be re-added
-        // to its container so we don"t need to manually add it.
-        // For more information, see the Fragments API guide at:
-        //
-        // http://developer.android.com/guide/components/fragments.html
-        //
+        // --
         if (savedInstanceState == null) {
             // Create the detail fragment and add it to the activity
             // using a fragment transaction.
             val fragment = ItemDetailFragment().apply {
                 arguments = Bundle().apply {
-                    putString(ItemDetailFragment.ARG_ITEM_ID,
-                            intent.getStringExtra(ItemDetailFragment.ARG_ITEM_ID))
+                    putSerializable(ItemDetailFragment.ARG_ITEM, character)
```

Comment:
In this very specific case, it is ok to send up the whole object. In complexes objects and if the detailed object requires further api calls, the id must be the only value to send

Reply:
I understand, but in this specific case, the list of items are the same ot detail item, so its no neccesary use another endpoint

For dama memory leaks there its no problem, a data object with Serializable parent extension has the avility to be Super Efficient